.circleci/config.yml

```diff
@@ -57,6 +57,33 @@ jobs:
       - run:
           name: Publish @next
           command: npm publish --tag next --access public
+  docker_next:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/cli-microsoft365
+      - setup_remote_docker:
+          version: 19.03.13
+      - run:
+          name: Extract version from package
+          command: |
+            echo "export PACKAGE_VERSION=$(node -p "require('./package.json').version")" >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Login to Docker Hub
+          command: docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
+      - run: 
+          name: Build :$PACKAGE_VERSION
+          command: docker build --pull --rm -t m365pnp/cli-microsoft365:$PACKAGE_VERSION "." --build-arg CLI_VERSION=$PACKAGE_VERSION
+      - run: 
+          name: Publish :$PACKAGE_VERSION
+          command: docker push m365pnp/cli-microsoft365:$PACKAGE_VERSION
+      - run: 
+          name: Build :next
+          command: docker build --pull --no-cache --rm -t m365pnp/cli-microsoft365:next "." --build-arg CLI_VERSION=next
```

Comment:
If we're executing two docker build commands one after another, couldn't we benefit of re-using the previously retrieved layers to speed up the build? There is no difference between the image tagged with the commit hash and `@next` other than the tag. By reusing the previously pulled layers, building the second image would be down to a second or two.

Reply:
Good observation, yes we can absolutely do that 🚀

Reply:
Updated in latest push